onnxscript/rewriter/rules/common/_fuse_batchnorm.py

```diff
@@ -68,7 +68,9 @@ def rewrite(self, op, x: ir.Value, inbound_out: ir.Value, batchnorm_out: ir.Valu
             bias_name = inbound_node.inputs[2].name
         else:
             original_bias = np.zeros_like(input_mean)
-            bias_name = x.name + "_bias"
```

Comment:
Curious why the original would collide?

Reply:
Sounds like:
1. When we create initializers, we need a way to create new unique names (at least unique among initializers, to begin with). I wonder if we can expose an IR method that will change an initializer's name to be unique (instead of throwing an exception) when adding it?
2. In this example, it should work fine since the first inputs are different. What happens if the first inputs are the same in two nodes? Granted, it won't happen in realistic examples. But it could still cause an exception, I think?

Reply:
> When we create initializers, we need a way to create new unique names (at least unique among initializers, to begin with). I wonder if we can expose an IR method that will change an initializer's name to be unique (instead of throwing an exception) when adding it?

Agree. In the created case, from what I saw, we don't even see an exception, the fused bias from pattern 1 and 2 co-exists with the same names.

> 2. In this example, it should work fine since the first inputs are different. What happens if the first inputs are the same in two nodes? Granted, it won't happen in realistic examples. But it could still cause an exception, I think?

In terms of Conv node, the example is problematic when bias is fused from op.BatchNorm to op.Conv because they will share the same naming.

Reply:
Feel free to continue the discussion